xcodeproj/internal/extra_flags_bazelrc.bzl

```diff
@@ -0,0 +1,82 @@
+"""Implementation of the `extra_flags_bazelrc` rule."""
+
+load("@bazel_skylib//rules:common_settings.bzl", "BuildSettingInfo")
+
+def _process_extra_flags(*, attr, content, setting, config_suffix):
+    extra_flags = getattr(attr, setting)[BuildSettingInfo].value
+    if extra_flags:
+        content.append(
+            "build:rules_xcodeproj{} {}".format(config_suffix, extra_flags),
+        )
+
+def _extra_flags_bazelrc_impl(ctx):
+    output = ctx.actions.declare_file("xcodeproj_extra_flags.bazelrc")
+
+    content = []
+
+    _process_extra_flags(
+        attr = ctx.attr,
+        content = content,
+        setting = "_extra_common_flags",
+        config_suffix = "",
+    )
+    _process_extra_flags(
+        attr = ctx.attr,
+        content = content,
+        setting = "_extra_generator_flags",
+        config_suffix = "_generator",
+    )
```

Comment:
I forgot that this will cause issues with CI. I'll pull this one out to be in the runner template itself.